src/array/ops/downcast.rs

```diff
@@ -3,43 +3,69 @@ use arrow2::array;
 
 use crate::{
     array::{BaseArray, DataArray},
-    datatypes::{BinaryArray, BooleanArray, DaftNumericType, DateArray, Utf8Array},
+    datatypes::{
+        BinaryArray, BooleanArray, DaftNumericType, DateArray, FixedSizeListArray, Utf8Array,
+    },
 };
 
-impl<T> DataArray<T>
+pub trait Downcastable {
+    type Output;
+
+    fn downcast(&self) -> &Self::Output;
```

Comment:
I was thinking that it would be nicer for implementers to specify the plain type, keeping the `downcast()` output type being a reference as a separate detail, but in hindsight that doesn't make much sense: the trait implementers specified `Output` type should be the verbatim `downcast()` output type, since the former isn't used for anything else.

I'll make that change!

Reply:
Ah and another reason is that throwing the reference on the `type Output = ` definition then requires defining an explicit lifetime...